lib/Config/ServicesChecker.php

```diff
@@ -132,6 +132,23 @@ public function isMailPoetAPIKeyPendingApproval(): bool {
     return $mssActive && $mssKeyValid && $mssKeyPendingApproval;
   }
 
+  public function isUserActivelyPaying(): bool {
+    $isPremiumKeyValid = $this->isPremiumKeyValid(false);
+    $premiumSupportTier = $this->settings->get('premium.premium_key_state.data.support_tier');
+    $isUserPayingForPremium = $premiumSupportTier === 'premium';
```

Comment:
There is already a method for checking this: `$this->subscribersFeature->hasPremiumSupport()`. It's worth keeping such logic in one place by reusing the existing method.

I know that `ServicesChecker` and `SubscribersFeature` have some duplicate checks for legacy reasons. Originally `ServicesChecker` was used for handling license-related error notices, and `SubscribersFeature` for checking if the subscriber limit is reached, but over time their functionality started to overlap as both needed more granular checks for various license-related cases. Ripe for refactoring, it seems.

Reply:
Noted.
I wasn’t aware of that method
Thanks for the review @wxa  🙌🏿